### ISSUE TITLE
Compiler FE: assert on ConvertNCHWToNHWC

### ISSUE THREAD

**Issue by seanshpark** [Samsung/ONE#5709]
Fails on assert in `compiler/luci/pass/src/ConvertNCHWToNHWCPass.cpp`

```cxx
      ConvertNCHWToNHWC converter;
      auto circle_node = loco::must_cast<luci::CircleNode *>(node);
      assert(circle_node->rank() == 4); <-- HERE
      if (circle_node->accept(&converter))
```

Test model: `build/compiler/luci/tests/Select_002.circle` or any other node with rank != 4

**Comment by seanshpark** [comment]
CC @jinevening , @parjong

**Comment by jinevening** [comment]
I don't know why assert was used there..  :confused: Thanks for the finding.

I uploaded a PR to fix it #5710. PTAL

**Comment by seanshpark** [comment]
Fixed!
